sled-agent/src/zone_bundle.rs

```diff
@@ -1676,6 +1668,98 @@ mod tests {
+        // Each OS implements slightly different `du` options.
+        //
+        // Linux and illumos support the "apparent" size in bytes, though using
+        // different options. macOS doesn't support bytes at all, and has a minimum
+        // block size of 512.
+        //
+        // We'll suffer the lower resolution on macOS, and get higher resolution on
+        // the others.
+        cfg_if::cfg_if! {
+            if #[cfg(target_os = "illumos")] {
+                const BLOCK_SIZE: u64 = 1;
+                const DU_ARG: &str = "-A";
+            } else if #[cfg(target_os = "linux")] {
+                const BLOCK_SIZE: u64 = 1;
+                const DU_ARG: &str = "-b";
+            } else if #[cfg(target_os = "macos")] {
+                const BLOCK_SIZE: u64 = 512;
+                const DU_ARG: &str = "-k";
+            } else {
+                compile_error!("unsupported target OS");
+            }
+        }
+
+        async fn disk_usage_du(path: &Utf8PathBuf) -> Result<u64, BundleError> {
+            const DU: &str = "du";
+            let args = &[DU_ARG, "-s", path.as_str()];
+            let output =
+                Command::new(DU).args(args).output().await.map_err(|err| {
+                    BundleError::Command {
+                        cmd: format!("{DU} {}", args.join(" ")),
+                        err,
+                    }
+                })?;
+            let err = |msg: &str| {
+                BundleError::Cleanup(anyhow!(
+                    "failed to fetch disk usage for {}: {}",
+                    path,
+                    msg,
+                ))
+            };
+            if !output.status.success() {
+                return Err(err("du command failed"));
+            }
+            let Ok(s) = std::str::from_utf8(&output.stdout) else {
+                return Err(err("non-UTF8 stdout"));
+            };
+            let Some(line) = s.lines().next() else {
+                return Err(err("no lines in du output"));
+            };
+            let Some(part) = line.trim().split_ascii_whitespace().next() else {
+                return Err(err("no disk usage size computed in output"));
+            };
+            part.parse()
+                .map(|x: u64| x.saturating_mul(BLOCK_SIZE))
+                .map_err(|_| err("failed to parse du output"))
+        }
+
+        let path =
+            Utf8PathBuf::from(concat!(env!("CARGO_MANIFEST_DIR"), "/src"));
+        let t0 = std::time::Instant::now();
+        let usage = dbg!(disk_usage(&path).await)
+            .expect("disk usage for src dir failed");
+        eprintln!("disk_usage({path}) took {:?}", t0.elapsed());
+
+        let t0 = std::time::Instant::now();
+        let du_usage =
+            dbg!(disk_usage_du(&path).await).expect("running du failed!");
+        eprintln!("du -s {path} took {:?}", t0.elapsed());
+
+        // Round down the Rust disk usage result to `du`'s block size on this
```

Comment:
Nit, you say "round down" here and "rounding up" in the message on stderr. I think it's the latter, based on the use of `div_ceil()`, right? Could we make the two match?